ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs

```diff
@@ -69,8 +69,7 @@ import           Ouroboros.Consensus.Util.Versioned
 import qualified Cardano.Ledger.BaseTypes as SL (ActiveSlotCoeff, Seed)
 import           Cardano.Ledger.Crypto (StandardCrypto, VRF)
 import qualified Shelley.Spec.Ledger.API as SL
-import qualified Shelley.Spec.Ledger.BlockChain as SL (checkLeaderValue, mkSeed,
-                     seedEta, seedL)
+import qualified Shelley.Spec.Ledger.BlockChain as SL (mkSeed, seedEta, seedL)
```

Comment:
So `checkLeaderValue` was unused? I wonder why that hadn't triggered a wanring.

Reply:
It is used [here](https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Protocol.hs#L507) but:
- `Shelley.Spec.Ledger.BlockChain` exports it.
- `Shelley.Spec.Ledger.API.Protocol` re-exports it.
- `Shelley.Spec.Ledger.API` re-exports all from `Shelley.Spec.Ledger.API.Protocol`.
- `Shelley.Spec.Ledger.API` is already imported qualified as `SL`.

So GHC claims it is a redundant import.